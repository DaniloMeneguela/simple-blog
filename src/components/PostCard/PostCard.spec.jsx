import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('Should render PostCard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByAltText(/title 1/i)).toHaveAttribute('src', 'img/img.png');

    expect(screen.getByRole('heading', { name: /title 1/i })).toBeInTheDocument();

    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
