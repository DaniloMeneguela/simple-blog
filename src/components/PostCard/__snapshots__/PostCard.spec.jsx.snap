// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h3>
       
      1
       
    </h3>
    <h2>
       
      title 1
       
    </h2>
    <p>
       
      body 1
       
    </p>
  </div>
</div>
`;
